Formula/geometry.rb

```diff
@@ -0,0 +1,37 @@
+class Geometry < Formula
+  desc "Minimal, fully customizable and composable zsh prompt theme"
+  homepage "https://github.com/geometry-zsh/geometry"
+  url "https://github.com/geometry-zsh/geometry/archive/v2.0.0.tar.gz"
+  sha256 "47c9fd345e5e8f093ae735a91fe5a9a9a6e3c1873eab220a7846a17b8405de02"
+  license "ISC"
+  head "https://github.com/geometry-zsh/geometry.git", branch: "main"
+
+  depends_on "zsh-async"
+  uses_from_macos "expect" => :test
+  uses_from_macos "zsh" => :test
+
+  def install
+    libexec.install ["functions", "geometry.zsh"]
```

Comment:
```suggestion
    pkgshare.install ["functions", "geometry.zsh"]
```
This seems more typical of other Zsh plugins.

Reply:
Really, all `prompt` themes should be `zsh_function.install`ed so they can be `autoload`ed for `promptinit`, but this one doesn't work…

Reply:
Yes, I don't find `pkgshare` a great place either (some `zsh_*` location is probably better), but that's where a bunch of other formula install `*.zsh` plugins, so we should at least try to be consistent.

Reply:
> we should at least try to be consistent

Agreed, but looking at other `prompt` formulae, it is currently very inconsistent… Some of them use `#{HOMEBREW_PREFIX}/share/` (would `opt` not be more appropriate here, anyway?), others `#{opt_prefix}/`, others [`opt_`]`pkgshare/`, and others still `zsh_function`.

Some have `caveats`, but others do not… It all seems a bit arbitrary.

For `zsh` `prompt`s specifically, `autoload`ing them from `#{prefix}/share/zsh/site-functions` (the `zsh_function.install` location, already found on`FPATH`) is the _proper_ way to install a `prompt` theme, and should speed up `zsh`ell initial load time, as opposed to just `source`ing directly in `.zshrc`…